### PR TITLE
Add support for .withsshcertificateauthenticationscheme when WAM enabled

### LINKED ISSUE #4909: [Bug] WithSSHCertificateAuthenticationScheme Not Return a SSH Certificate When WAM Enabled
### Library version used

4.61.3.0

### .NET version

.netstandard 2.0

### Scenario

PublicClient - desktop app

### Is this a new or an existing app?

The app is in production, and I have upgraded to a new version of MSAL

### Issue description and reproduction steps

Raised by @vthiebaut10, in Azure PowerShell, when WAM is enabled,  the customer fails to retry to connect a configured Azure VM using Az.Ssh.
Here are the steps to reproduce
- Create an Ubuntu Azure VM with Login with Microsoft Entra Id enabled ([Sign in to a Linux virtual machine in Azure by using Microsoft Entra ID and OpenSSH - Microsoft Entra ID | Microsoft Learn](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fentra%2Fidentity%2Fdevices%2Fhowto-vm-sign-in-azure-ad-linux&data=05%7C02%7CLei.Jin%40microsoft.com%7C1d653c7c689b4c3bdd4408dcc1f496c9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638598503517380022%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=ow1MX2F44CEUQeLtMoxs2vUaINRU6lWL5ZVVuQkuEgU%3D&reserved=0)). To enable AAD login, during creation in the portal, on the management tab select the Login with Microsoft Entra ID checkbox and ensure that the System assigned managed identity checkbox is selected.
- Once the VM is created, assign "Virtual Machine Administrator Login" role to your account.
- Run "Enter-AzVM -ResourceGroupName rgname -Name vmname" (make sure wam is enabled)

When the WAM is disabled, Enter-AzVM works as expected.
And we find when WAM is enabled, WithSSHCertificateAuthenticationScheme doesn't return a SSH certificate but return a bearer token. But WAM is disabled, it returns a SSH certficate.
### Relevant code snippets

```csharp
public virtual IPublicClientApplication CreatePublicClient(string authority, string tenantId)
{
    var builder = PublicClientApplicationBuilder.Create(Constants.PowerShellClientId);
    if (AzConfigReader.IsWamEnabled(authority))
    {
        builder = builder.WithBroker(new BrokerOptions(BrokerOptions.OperatingSystems.Windows));
    }
    if (!string.IsNullOrEmpty(authority))
    {
        builder.WithAuthority(authority, tenantId ?? organizationTenant);
    }
    var client = builder.Build();
    RegisterCache(client);
    return client;
}

var publicClient = CreatePublicClient(context.Environment.ActiveDirectoryAuthority, context.Tenant.Id)
var account = publicClient.GetAccountAsync(context.Account.GetProperty(AzureAccount.Property.HomeAccountId))
               .ConfigureAwait(false).GetAwaiter().GetResult();
var result = publicClient.AcquireTokenSilent(scopes, account)
             .WithSSHCertificateAuthenticationScheme(jwk, keyId)
             .ExecuteAsync();
var accessToken = result.ConfigureAwait(false).GetAwaiter().GetResult();
```

### Expected behavior

Return a SSH certificate as non-broker public client does

### Identity provider

Microsoft Entra ID (Work and School accounts and Personal Microsoft accounts)

### Regression

_No response_

### Solution and workarounds

_No response_

### PR DESCRIPTION
**Pull request by Ugonnaak1**
Fixes #4909 

**Changes proposed in this request**

.withsshcertificateauthenticationscheme currently does not return a ssh certificate when WAM is enabled. This change involves relaying relevant parameters to MsalRuntime in order to receive a ssh certificate 
**Testing**

Testing done using manual validation. Tests also added for silent and interactive flows
Test app updated to support interactive flow for ssh.

**Performance impact**

**Documentation**
- [ ] All relevant documentation is updated.

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/RuntimeBrokerTests.cs [hunk 1]
```diff
@@ -223,6 +243,47 @@ await AssertException.TaskThrowsAsync<MsalUiRequiredException>(
                 .ConfigureAwait(false);
         }
 
+        [IgnoreOnOneBranch]
+        [TestMethod]
+        public async Task WamWithSSHCertificateAuthenticationSchemeAsync()
```
Comment (iulico-1) [inline tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/RuntimeBrokerTests.cs:248]: >WamWithSSHCertificateAuthenticationSchemeAsync [](http://example.com/codeflow?start=26&length=46)

Thank you for adding this test coverage. Does these tests run in CI ?

**Review by iulico-1 — APPROVED** [review]
:shipit:

**Review by bgavrilMS — APPROVED** [review]

File: tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/RuntimeBrokerTests.cs (hunk 1 above)
Comment (Ugonnaak1) [reply tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/RuntimeBrokerTests.cs:248]: Yes

File: tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/RuntimeBrokerTests.cs [hunk 2]
```diff
@@ -223,6 +243,50 @@ await AssertException.TaskThrowsAsync<MsalUiRequiredException>(
...
+        {
+            IntPtr intPtr = GetForegroundWindow();
+            Func<IntPtr> windowHandleProvider = () => intPtr;
+
+
+            var labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
```
Comment (ashok672) [inline tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/RuntimeBrokerTests.cs:254]: nit. Can you remove the extra new lines,

**Review by ashok672 — APPROVED** [review]
:shipit:

File: src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs [hunk 3]
```diff
@@ -181,6 +181,16 @@ public static NativeInterop.AuthParameters GetCommonAuthParameters(
             {
                 authParams.Properties["instance_aware"] = "true";
             }
+            
+            //SSH Cert
+            if(authenticationRequestParameters.AuthenticationScheme.AccessTokenType == "ssh-cert")
```
Comment (iulico-1) [inline src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs:186]: >  if(authenticationRequestParameters.AuthenticationScheme.AccessTokenType == "ssh-cert") [](http://example.com/codeflow?start=10&length=88)

Should this be a case-insensitive check ?

Comment (Ugonnaak1) [reply src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs:186]: From what I've seen, the AccessTokenType is always lowercase

**Review by bgavrilMS — APPROVED** [review]

**Pull request merged**